dashboard/app/controllers/teacher_dashboard_controller.rb

```diff
@@ -1,6 +1,11 @@
 class TeacherDashboardController < ApplicationController
   def index
```

Comment:
the routes in this controller seem a bit non-standard -- I would recommend renaming `index` to `show` to more closely follow rails conventions. ok to do in a separate PR, although now might be a good time since you're adding a bunch of references to this route in the unit test.